### ISSUE TITLE
Taxonomy CRUD API should compute `active` property based on ancestors' `active` property

### ISSUE THREAD

**Issue by adamsachs** [ethyca/fides#3846]
### Is your feature request related to a specific problem?

In https://github.com/ethyca/fides/pull/3784 we added an `active` property to taxonomy elements (in our API and DB models). The desired behavior of this functionality (generally covered in https://github.com/ethyca/fides/issues/3710) is such that we'd like _descendants_ (i.e. children, granchildren, etc.) of a deactivated taxonomy element to also show up as effectively deactivated to the FE.

### Describe the solution you'd like

We will need to compute on the BE the _effective_ `active` property of a given taxonomy element whenever that element is returned by the API. The _effective_ `active` value based on both its own `active` property value in the DB as well as its ancestors' `active` property values - all must be `true` for an elements' effective `active` property to also be `true`.

We should compute this at the API layer, and _not_ at the DB level -- i.e. as "late" as possible -- to avoid:
- additional complexity related to propagating state across elements as ancestor taxonomy elements change state
    - relatedly: not computing the property at the DB level also helps us avoid more special-cased logic in our update operations - all the special-casing for the `active` property can just be done when elements are being retrieved/returned by the API.
- to maintain information about which taxonomy elements have been _explicitly_ marked as active/inactive by users. i.e. if a parent element changes from `active=false` to `active=true` state, we need to know whether a child had been explicitly marked as `active=false` to know what its new _effective_ `active` property should be

### Describe alternatives you've considered, if any

We also discussed computing the "effective" `active` state on the FE, which would be feasible. But we judged that to introduce more unneeded logic and complexity on the FE, so computing this on the BE was preferred.

### Additional context

See [related slack thread](https://ethyca.slack.com/archives/C04S7K0T2V7/p1689861844663849) for some more context.

**Comment by rsilvery** [comment]
@adamsachs is this still an issue?

**Comment by adamsachs** [comment]
> @adamsachs is this still an issue?

yup, assuming we still consider that the desired behavior here is to deactivate children/descendants of a parent taxonomy element when the parent is deactivated. that's not what currently happens, and this ticket is meant to cover that gap between current functionality and desired behavior, which still exists afaik.
